frontend/src/reducer.test.ts

```diff
@@ -0,0 +1,136 @@
+import { nanoid } from 'nanoid';
```

Comment:
most of this is just moved, scroll down to comments to see actually changed code